src/main/java/nl/uu/cs/ape/utils/APEFiles.java

```diff
@@ -333,82 +333,104 @@ private static File readURLToFile(String fileUrl) throws IOException {
     }
 
     /**
-     * Append text to the existing file. It adds the text at the end of the content
+     * Appends text to the existing file. It adds the text at the end of the content
      * of the file.
-     * 
-     * @param file    - existing file
-     * @param content - content that should be appended
-     * @throws IOException          in case of an I/O error
-     * @throws NullPointerException if the file is null
+     *
+     * @param file    The existing file.
+     * @param content The content that should be appended.
+     * @throws IOException          In case of an I/O error.
+     * @throws NullPointerException If the file or content is null.
      */
-    public static void appendToFile(File file, String content) throws IOException, NullPointerException {
-        Writer fileWriter = new FileWriterWithEncoding(file, "ASCII", true);
-        try (BufferedWriter writer = new BufferedWriter(fileWriter, 8192 * 4)) {
+    public static void appendToFile(File file, String content) throws IOException {
+        if (file == null || content == null) {
+            throw new NullPointerException("File or content cannot be null.");
+        }
+
+        try (BufferedWriter writer = new BufferedWriter(
+                new OutputStreamWriter(new FileOutputStream(file, true), StandardCharsets.US_ASCII), 8192 * 4)) {
             writer.write(content);
         }
     }
 
     /**
-     * Append text to the existing file. It adds the text at the end of the content
-     * of the file.
-     * 
-     * @param file    - existing file
-     * @param content - content that should be appended
-     * @throws IOException          in case of an I/O error
-     * @throws NullPointerException if the file is null
+     * Appends a set of text strings to the existing file. Each element in the set
+     * is added at the end of the content of the file.
+     *
+     * @param file    The existing file.
+     * @param content The set of content strings that should be appended.
+     * @throws IOException          In case of an I/O error.
+     * @throws NullPointerException If the file or content set is null.
      */
-    public static void appendSetToFile(File file, Set<String> content) throws IOException, NullPointerException {
-        Writer fileWriter = new FileWriterWithEncoding(file, "ASCII", true);
-        try (BufferedWriter writer = new BufferedWriter(fileWriter, 8192 * 4)) {
+    public static void appendSetToFile(File file, Set<String> content) throws IOException {
+        if (file == null || content == null) {
+            throw new NullPointerException("File or content set cannot be null.");
+        }
+
+        try (BufferedWriter writer = new BufferedWriter(
```

Comment:
The writer is instantiated in this method, appendToFile (base case) and appendToFile (CNFClause case) identically. Just an idea: If you implement a function which opens the file (if it exists and excepts otherwise) and writes a list of strings in append mode to the file, you could reuse that function and restrict the current functions to checking whether the encoding/content is non-null. Also, the current functions would have to provide a list of strings to the new function.